feat: Updates to HitPay Payment App

MuhammadAimanSulaiman opened the pull request:
## What does this PR do?
 This PR includes updates to the HitPay app

- New embed feature for cal.com embed 
- Update to the app description
- Fixes to customers being able to exceed bookings in a same timeslot when using HitPay

## Visual Demo (For contributors especially)

### Image Demo (if applicable):
 #### New embed feature for cal.com embed
  1. Run the cal.com embed script for an event with HitPay app installed.
  
![ebmed1](https://github.com/user-attachments/assets/ebc67c6a-68c0-470f-b99c-e01f6e5ba9d1)

  2. Schedule an event using the embedded booking form.
  
![embed2](https://github.com/user-attachments/assets/f03d9807-a5d2-4fb4-9ac7-6fa88f5645fe)
![embed3](https://github.com/user-attachments/assets/eb041eb4-f383-4061-aba4-bdfdf1f1432d)

  4.  The HitPay payment modal should appear.
![embed4](https://github.com/user-attachments/assets/ec84fde4-29c4-43fc-9c50-8608e5aebb61)
    * In the original version, the payment process fails.

  6. The payment should be completed successfully.
![embed5](https://github.com/user-attachments/assets/ca8f70c0-4177-4fa8-bff2-6502bc3d1eb7)
![embed6](https://github.com/user-attachments/assets/b056a75b-fb8e-4ce7-982e-412aeca3eecb)

 #### Update to the app description
  - The original description of the HitPay app is: 
![description_org](https://github.com/user-attachments/assets/429dfc87-afbc-4744-86a7-6488101e41b0)
  - The updated description is:
![description_new](https://github.com/user-attachments/assets/2e6a5b77-28d1-40c1-9f06-2772a2d81ca4)

 #### Fixes to customers being able to exceed bookings in the same timeslot when using HitPay
  1. Disable Offer seats 
     - Disable Offer Seats for an event with HitPay app installed.
     ![Screenshot from 2025-03-05 10-37-25](https://github.com/user-attachments/assets/ee626dd8-49a7-4d9c-bfc2-56af28d92961)

     - Attempt to schedule the same event simultaneously in two separate browsers using different email addresses.  
![Screenshot from 2025-03-05 10-41-57](https://github.com/user-attachments/assets/53f65b56-6a04-4bb1-a862-211942eaf4b8)

     - One of the booking attempts should fail.
    ![Screenshot from 2025-03-05 10-42-43](https://github.com/user-attachments/assets/2dd52a56-38e9-4748-8223-cd4407cc9cbe)
    * In the original version, HitPay payment was successfully processed for both bookings.
![Screenshot from 2025-03-05 11-57-28](https://github.com/user-attachments/assets/016828e0-7cad-43b4-9ef4-04022d4c9505)

  2. Enable Offer seats
   - Enable Offer Seats for an event with HitPay app installed.
   ![Screenshot from 2025-03-05 10-37-34](https://github.com/user-attachments/assets/c74588ce-a7ec-4478-9f75-a4a4b102d9bf)

    - Attempt to schedule the same event simultaneously in 3 separate browsers using different email addresses.
   ![Screenshot from 2025-03-05 10-49-17](https://github.com/user-attachments/assets/239b7319-444b-45fc-84ff-701963c479db)

    - One of the booking attempts should fail.
![Screenshot from 2025-03-05 10-50-15](https://github.com/user-attachments/assets/88e81bbd-785f-471a-b8e0-9ed1a57f9952)
 
    * In the original version, HitPay payment was successfully processed for both bookings.
![Screenshot from 2025-03-05 12-00-44](https://github.com/user-attachments/assets/a6bcd03e-cfc9-4373-ba04-56617d8656e9)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

## Checklist

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/calcom/cal.com?pullRequest=19737) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/calcom/cal.com?pullRequest=19737) before we can accept your contribution.<br/>**20** out of **21** committers have signed the CLA.<br/><br/>:white_check_mark: luzpaz<br/>:white_check_mark: Udit-takkar<br/>:white_check_mark: anikdhabal<br/>:white_check_mark: Ryukemeister<br/>:white_check_mark: vijayraghav-io<br/>:white_check_mark: harshit078<br/>:white_check_mark: steven-tey<br/>:white_check_mark: hbjORbj<br/>:white_check_mark: rolznz<br/>:white_check_mark: ddoemonn<br/>:white_check_mark: keithwillcode<br/>:white_check_mark: sn10-debug<br/>:white_check_mark: eunjae-lee<br/>:white_check_mark: supalarry<br/>:white_check_mark: emrysal<br/>:white_check_mark: MuhammadAimanSulaiman<br/>:white_check_mark: cnhhoang850<br/>:white_check_mark: SySagar<br/>:white_check_mark: nayan-bagale<br/>:white_check_mark: asadath1395<br/>:x: calcom-bot<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/calcom/cal.com?pullRequest=19737) it.</sub>

PeerRich commented:
this PR looks very odd, did you rebase incorrectly by chance? 

we are running into a lot of merge conflicts

![CleanShot 2025-03-05 at 10 07 03@2x](https://github.com/user-attachments/assets/79bdfe13-5b1f-44a1-8be3-739826fbf0ae)

MuhammadAimanSulaiman commented:
Hello @PeerRich 
I will resolve the conflicts soon

MuhammadAimanSulaiman committed:
feat: add hitpay embed feature for cal embed

feat: update hitpay app description

fix: fixed customers being able to exceed bookings in the same timeslot when using HitPay

fix: fixed callback issue in cal.com using hitpay app when not receiving payment responses

MuhammadAimanSulaiman commented:
Hello @PeerRich 
I resolved all the conflicts

MuhammadAimanSulaiman committed:
fix: fixed type-check error

anikdhabal reviewed (approved):
Thanks @MuhammadAimanSulaiman
